components/scream/src/control/atmosphere_driver.cpp

```diff
@@ -226,29 +222,29 @@ void AtmosphereDriver::finalize ( /* inputs? */ ) {
 }
 
 void AtmosphereDriver::register_groups () {
-  using ci_string = ekat::CaseInsensitiveString;
-  using ci_string_pair = std::pair<ci_string,ci_string>;
+  using GroupRequest = AtmosphereProcess::GroupRequest;
+
+  // Given a list of group-grid pairs (A,B), make sure there is a copy
+  // of each field in group A on grid B registered in the repo.
+  auto lambda = [&](const std::set<GroupRequest>& groups_grids) {
```

Comment:
Maybe you could call this `has_registered_field_copy` (or something) instead of `lambda`, given how you've named the lambda helper below.